### ISSUE TITLE
MultiPartUpload response content type

### ISSUE THREAD

**Issue by andemi** [gaul/s3proxy#320]
Hello, 
version 1.7.0 has the same error as here (version 1.6.1 doesn't have)
https://github.com/adobe/S3Mock/issues/35
https://github.com/adobe/S3Mock/issues/210

> 2020-02-25 16:08:50.149 WARN  HttpChannel - /input_9c2Ex/bucketKey.json?uploadId=323aaab8-c622-4489-befe-1dc260385bc7 [S3Proxy-Jetty-29|]
java.lang.NullPointerException: null
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:882)
        at com.google.common.io.BaseEncoding$StandardBaseEncoding.trimTrailingPadding(BaseEncoding.java:676)
        at com.google.common.io.BaseEncoding.decodeChecked(BaseEncoding.java:231)
        at com.google.common.io.BaseEncoding.decode(BaseEncoding.java:217)
        at org.jclouds.blobstore.config.LocalBlobStore.completeMultipartUpload(LocalBlobStore.java:843)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.google.inject.internal.DelegatingInvocationHandler.invoke(DelegatingInvocationHandler.java:37)
        at com.sun.proxy.$Proxy47.completeMultipartUpload(Unknown Source)
        at org.gaul.s3proxy.S3ProxyHandler$3.run(S3ProxyHandler.java:2288)
2020-02-25 16:08:50.149 WARN  HttpChannel - Could not send response error 500: java.lang.NullPointerException [S3Proxy-Jetty-29|]
[info] HistorySpec *** ABORTED ***
[info]   org.xml.sax.SAXParseException: XML document structures must start and end within the same entity.
[info]   at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:203)
[info]   at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:177)
[info]   at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400)
[info]   at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327)
[info]   at com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(XMLScanner.java:1472)
[info]   at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.endEntity(XMLDocumentFragmentScannerImpl.java:899)
[info]   at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.endEntity(XMLDocumentScannerImpl.java:559)
[info]   at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.endEntity(XMLEntityManager.java:1398)
[info]   at com.sun.org.apache.xerces.internal.impl.XMLEntityScanner.load(XMLEntityScanner.java:1916)
[info]   at com.sun.org.apache.xerces.internal.impl.XMLEntityScanner.skipSpaces(XMLEntityScanner.java:1655)
[info]   ...

**Comment by timuralp** [comment]
@andemi can you add the request that was made? Did the complete multipart upload call include the ETags for each part?

**Comment by andemi** [comment]
s3proxy in filesystem mode for Windows
Request in scala code
`  val file: Source[ByteString, NotUsed] =
      Source.single(ByteString(
        """
         {
            "timeRange": {
              "from": "2019-12-31",
              "to": "2020-01-01"
            }
          }
          """))


`    val s3Sink: Sink[ByteString, Future[MultipartUploadResult]] = S3
    .multipartUpload(inputBucket, "bucketKey.json")
    .withAttributes(S3Attributes.settings(s3Input.useInputSettings))
`

> Did the complete multipart upload call include the ETags for each part?
I don't know about each part

s3proxy v1.6.1 result is:

`MultipartUploadResult(location=http://example-bucket.s3.amazonaws.com/data/bucketKey.json,bucket=input,key=data/bucketKey.json,etag=e7d4be8ee4e7234f534c05e13ccc593d,versionId=None)`


In transient mode for v1.7.0 s3proxy has no error

**Comment by timuralp** [comment]
From the error it looks like jclouds doesn't have the ETag associated with the individual parts. I'll try to reproduce the behavior on Windows -- it might be related to not having extended attributes.

**Comment by timuralp** [comment]
@andemi I was able to reproduce this issue. It's not windows-specific, but shows up any time s3proxy is used against a filesystem without extended attributes support (e.g. FAT). I submitted a pull request to jclouds to remedy this (https://github.com/jclouds/jclouds/pull/1285). In the meantime, I tested that completing MPUs works correctly on NTFS. If possible, I would consider running s3proxy against that or another filesystem with support for extended attributes.
